pkg/reactable/catalyze.go

```diff
@@ -0,0 +1,109 @@
+package reactable
+
+import (
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+)
+
+func (r *Reactable) tryAggravate(a *combat.AttackEvent) {
+	if a.Info.Durability < ZeroDur {
+		return
+	}
+
+	switch a.Info.Element {
```

Comment:
this check isn't necessary since you checked in the switch statement already.

the other ones have it because they can be triggered from multiple source elements

Reply:
The other reactions check to ensure that the trigger is correct. I think this should still do that for consistency